internal/ethapi/api.go

```diff
@@ -1165,10 +1165,31 @@ func AccessList(ctx context.Context, b Backend, blockNrOrHash rpc.BlockNumberOrH
 	// Retrieve the precompiles since they don't need to be added to the access list
 	precompiles := vm.ActivePrecompiles(b.ChainConfig().Rules(header.Number, isPostMerge, header.Time))
 
+	// addressesToExclude contains sender, receiver, precompiles and valid authorizations
+	addressesToExclude := map[common.Address]struct{}{args.from(): {}, to: {}}
+	for _, addr := range precompiles {
+		addressesToExclude[addr] = struct{}{}
+	}
+
+	// Prevent redundant operations if args contain more authorizations than EVM may handle
+	maxAuthorizations := uint64(*args.Gas) / params.TxAuthTupleGas
+	if uint64(len(args.AuthorizationList)) <= maxAuthorizations {
```

Comment:
I think it's a good idea to limit the number of authorizations since they incur a heavy signature verification cost. I would instead of changing semantics rather return an error once the cap is reached.

Let's use CallNewAccountGas for the max number